python/pyspark/errors/error-classes.json

```diff
@@ -0,0 +1,7 @@
+{
+  "COLUMN_IN_LIST" : {
+    "message" : [
+      "<funcName> does not allow a column in a list"
+    ]
```

Comment:
Will add more error-classes in follow-up PRs.

Since this PR introduces the error class for PySpark for the first time, I made only one example for the convenience of review.

Reply:
Is this the standard way to put other files? Now it looks weird that `pyspark.errors` package contains JSON files. Adding any reference or docs related to this would be helpful to review futher.

Reply:
Thanks for the comment, @HyukjinKwon .

I investigated some reference.

According to  [Installing Additional Files](https://docs.python.org/3.10/distutils/setupscript.html#installing-additional-files ) in Python documentation, seems like we can put the additional files for handling image, config, etc by using `data_files` option when installing the Python project.

Also, [Adding Non-Code Files for python packaging](https://python-packaging.readthedocs.io/en/latest/non-code-files.html#adding-non-code-files) shows an example that using JSON file when packaging the Python.

I think we can't say it's absolute standard way, but maybe we can say at least it's generally accepted way in Python

Although I haven't see the similar Python project or standard that using JSON file as a format for error class, I think using a json file as the error class format is fine as it is consistent within the Apache Spark project, IMHO.